src/ngraph/runtime/tensor.hpp

```diff
@@ -103,6 +103,12 @@ namespace ngraph
             /// \param n Number of bytes to read, must be integral number of elements.
             virtual void read(void* p, size_t n) const = 0;
 
+            /// \brief check tensor for new data, call may block.
+            ///    backends may use this to ensure tensor is updated (eg: lazy eval).
+            virtual void wait_to_read() {}
```

Comment:
How about `wait_for_read_ready` and `wait_for_write_ready`

Reply:
yes sounds more descriptive, will make the change.